pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1559,38 +1614,6 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 			return err
 		}
 
-		if useIOThreads {
-			bus := getBusFromDisk(disk)
-			switch bus {
-			// Only disks with virtio bus support IOThreads
-			case v1.DiskBusVirtio:
-				ioThreadId := defaultIOThread
-				dedicatedThread := false
-				if disk.DedicatedIOThread != nil {
-					dedicatedThread = *disk.DedicatedIOThread
-				}
-
-				if dedicatedThread {
-					ioThreadId = currentDedicatedThread
-					currentDedicatedThread += 1
-				} else {
-					ioThreadId = currentAutoThread
-					// increment the threadId to be used next but wrap around at the thread limit
-					// the odd math here is because thread ID's start at 1, not 0
-					currentAutoThread = (currentAutoThread % uint(autoThreads)) + 1
-				}
-				newDisk.Driver.IOThread = &ioThreadId
-
-			// We can find a workaround by adding IOThreads to the SCSI controller
-			case v1.DiskBusSCSI:
-				setIOThreadSCSIController = true
-
-			default:
-				// TODO: if possible, find a workaround for SATA and USB buses.
-				log.Log.Infof("IOthreads not available for disk %s: Bus %s not supported", disk.Name, bus)
-			}
-		}
-
```

Comment:
This is not equivalent.
In this block the `newDisk.Driver.IOThread` can be changed.
The `newDisk` is later used at line 1597.
Am I missing something?

Reply:
Oh maybe I got it. It is set later in the setIOThread. Ignore it

Reply:
@fossedihelm if I understand the comment, yes, this is moved in the setIOThreads

Reply:
Yes, I see